server/src/internalClusterTest/java/org/elasticsearch/search/query/SearchQueryIT.java

```diff
@@ -2079,4 +2079,21 @@ public void testIssueFuzzyInsideSpanMulti() {
         SearchResponse response = client().prepareSearch("test").setQuery(query).get();
         assertHitCount(response, 1);
     }
+
+    public void testFetchIdFieldQuery() {
+        createIndex("test");
+        int docCount = randomIntBetween(10, 50);
+        for (int i=0; i<docCount; i++) {
+            client().prepareIndex("test").setSource("field", "foobarbaz").get();
+        }
+        ensureGreen();
+        refresh();
+
+        SearchResponse response = client().prepareSearch("test").addFetchField("_id").setSize(10).get();
```

Comment:
It might make sense to check all documents we indexed (docCount), not just top 10.